### PR TITLE
[core] Add interface to add images to the style

### PR DESCRIPTION
**Pull request by tmpsantos**
Unlike annotations, they will go away when we change the style. This API can also be used to change a sprite downloaded by a style at runtime.

### CODE REVIEW

File: include/mbgl/map/map.hpp [hunk 1]
```diff
@@ -161,6 +161,10 @@ class Map : private util::noncopyable {
     void addLayer(std::unique_ptr<style::Layer>, const optional<std::string>& beforeLayerID = {});
     void removeLayer(const std::string& layerID);
 
+    // Add icon, bound to the style
+    void addIcon(const std::string&, std::shared_ptr<const SpriteImage>);
```
Comment (jfirebaugh) [inline include/mbgl/map/map.hpp]: Should be `std::unique_ptr` for consistency with `addLayer` and `addSource`.

**Comment by tmpsantos** [comment]
@jfirebaugh Thanks for reviewing. Changes made.

> Want to start with that naming convention here?

Makes sense.

**Comment by 1ec5** [comment]
Tracking iOS/macOS support for style image management in #6436.

/cc @zugaldia

**Comment by zugaldia** [comment]
@1ec5 Thanks for the heads up. Android's work in now tracked in https://github.com/mapbox/mapbox-gl-native/issues/6459.

**Pull request merged**